RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -604,8 +604,16 @@ void EGExtraInfoModifierFromDB::modifyObject(reco::Photon& pho) const {
   if (iseb) {
     EBDetId ebseedid(theseed->seed());
     eval[26] = pho.e5x5()/theseed->energy();
-    eval[27] = ebseedid.ieta();
-    eval[28] = ebseedid.iphi();
+    int ieta = ebseedid.ieta();
+    int iphi = ebseedid.iphi();
+    eval[27] = ieta;
+    eval[28] = iphi;
+    eval[29] = (ieta-1*abs(ieta)/ieta)%5;
+    eval[30] = (iphi-1)%2;
+    eval[31] = (abs(ieta)<=25)*((ieta-1*abs(ieta)/ieta)%25) + (abs(ieta)>25)*((ieta-26*abs(ieta)/ieta)%20);
+    eval[32] = (iphi-1)%20;
+    eval[33] = ieta;  /// duplicated variables but this was trained like that
+    eval[34] = iphi;  /// duplicated variables but this was trained like that
   } else {
     EEDetId eeseedid(theseed->seed());
     eval[26] = the_sc->preshowerEnergy()/raw_energy;
```

Comment:
yes, only the EB was missing variables.

On 4 févr. 2016, at 23:38, Slava Krutelyov <notifications@github.com<mailto:notifications@github.com>> wrote:


In RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc<https://github.com/cms-sw/cmssw/pull/13199#discussion_r51949685>:

> @@ -604,8 +604,16 @@ void EGExtraInfoModifierFromDB::modifyObject(reco::Photon& pho) const {
>    if (iseb) {
>      EBDetId ebseedid(theseed->seed());
>      eval[26] = pho.e5x5()/theseed->energy();
> -    eval[27] = ebseedid.ieta();
> -    eval[28] = ebseedid.iphi();
> +    int ieta = ebseedid.ieta();
> +    int iphi = ebseedid.iphi();
> +    eval[27] = ieta;
> +    eval[28] = iphi;
> +    eval[29] = (ieta-1*abs(ieta)/ieta)%5;
> +    eval[30] = (iphi-1)%2;
> +    eval[31] = (abs(ieta)<=25)*((ieta-1*abs(ieta)/ieta)%25) + (abs(ieta)>25)*((ieta-26*abs(ieta)/ieta)%20);
> +    eval[32] = (iphi-1)%20;
> +    eval[33] = ieta;  /// duplicated variables but this was trained like that
> +    eval[34] = iphi;  /// duplicated variables but this was trained like that
>    } else {
>      EEDetId eeseedid(theseed->seed());
>      eval[26] = the_sc->preshowerEnergy()/raw_energy;


just to double-check: is the EE part correct? (no missing variables)

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/13199/files#r51949685>.